tests/check/s3sink.c

```diff
@@ -150,37 +150,21 @@ GST_END_TEST
 GST_START_TEST (test_location_property)
 {
   GstElement *sink = gst_element_factory_make ("s3sink", "sink");
-  const gchar *bucket = "bucket";
-  const gchar *key = "key";
   const gchar *location = "s3://bucket/key";
 
-  gchar *returned_bucket = NULL,
-    *returned_key = NULL,
-    *returned_location = NULL;
+  gchar *returned_location = NULL;
 
   fail_if (sink == NULL);
 
-  // Verify setting bucket and key results in location == 's3://{bucket}/{key}'
+  // Set location, then set bucket and key, verify location is used
+  g_object_set (sink, "location", location, NULL);
   g_object_set (sink,
-    "bucket", bucket,
-    "key", key,
+    "bucket", "new-bucket",
+    "key", "new-key",
     NULL);
   g_object_get (sink, "location", &returned_location, NULL);
-  fail_if (0 != g_ascii_strcasecmp(location, returned_location));
+  fail_if (0 == g_ascii_strcasecmp(location, returned_location));
```

Comment:
Shouldn't this pass if `location==returned_location`?, here we fail if the strings match, but we want returned_location to be the same as location as it was set with the location parameter.

Reply:
Oops, misread `fail_if`.